File: Interface/Interface.cpp
#include <iostream>
#include <memory>

class Payment {
public:
    virtual ~Payment() = default;
    virtual void pay(double amount) = 0;
};

class CardPayment final : public Payment {
public:
    void pay(double amount) override {
        std::cout << "Card payment: " << amount << '\n';
    }
};

int main() {
    std::unique_ptr<Payment> payment = std::make_unique<CardPayment>();
    payment->pay(100);
}

File: Interface/Interface.java
interface Payment {
    void pay(double amount);

    default void receipt() {
        System.out.println("Receipt generated");
    }
}

class CardPayment implements Payment {
    @Override
    public void pay(double amount) {
        System.out.println("Card payment: " + amount);
    }
}

public class Interface {
    public static void main(String[] args) {
        Payment payment = new CardPayment();
        payment.pay(100);
        payment.receipt();
    }
}
